pkgs/applications/misc/rofimoji/default.nix

```
@@ -0,0 +1,43 @@
+{ buildPythonApplication
+, fetchFromGitHub
+, lib
+
+, waylandSupport ? true
+, x11Support ? true
+
+, ConfigArgParse
+, pyxdg
+, rofi
+, wl-clipboard
+, wtype
+, xdotool
+, xsel
+}:
+
+buildPythonApplication rec {
+  pname = "rofimoji";
+  version = "4.3.0";
+
+  src = fetchFromGitHub {
+    owner = "fdw";
+    repo = "rofimoji";
+    rev = version;
+    sha256 = "08ayndpifr04njpijc5n5ii5nvibfpab39p6ngyyj0pb43792a8j";
+  };
+
+  propagatedBuildInputs = with lib; [ ConfigArgParse pyxdg rofi ]
+    ++ ifEnable waylandSupport [ wl-clipboard wtype ]
```

```suggestion
    ++ optionals waylandSupport [ wl-clipboard wtype ]
```

this is `lib/deprecated.nix` and it's rarely used
(optionals does basically the same thing)

Same for below this

---

Done.